aio_pika/queue.py

```diff
@@ -513,22 +528,61 @@ async def __aexit__(
     async def __anext__(self) -> IncomingMessage:
         if not hasattr(self, "_consumer_tag"):
             await self.consume()
+
+        if self._closed.is_set():
+            raise StopAsyncIteration
+
+        message = asyncio.create_task(
```

Comment:
Creating task on each message is significantly slower. Task creation works fine when the coroutine do some long job or waiting network.

Here, as many as three tasks are created that must not only be queued for execution, but also canceled upon the occurrence of one of the events.

Reply:
As of Python 3.11 passing coroutines to `asyncio.wait()` is forbidden. So I recuded it to one task per message, instead of three.

I don't think it is possible to remove that last task without effectively reimplementing `asyncio.wait()`.

Reply:
I was wrong.

No more task for message. In fact, I skipped the `await` if there are messages in the queue and use the `get_nowait` function instead.